src/config.js

```diff
@@ -1,14 +1,18 @@
 require('dotenv').config();
 
-function getWebsocketPulsarUrl() {
+function getWebsocketPulsarUrl(role='consumer') {
     const hostname = process.env.ADDON_PULSAR_HOSTNAME;
     const port = process.env.ADDON_PULSAR_HTTP_PORT;
     const tenant = process.env.ADDON_PULSAR_TENANT;
     const namespace = process.env.ADDON_PULSAR_NAMESPACE;
 
-    return `wss://${hostname}:${port}/ws/v2/producer/persistent/${tenant}/${namespace}`;
+    return `wss://${hostname}:${port}/ws/v2/${role}/persistent/${tenant}/${namespace}`;
 }
 
-module.exports.getWebsocketPulsarUrlWithTopic = function (topic) {
-    return getWebsocketPulsarUrl() + `/${topic}`;
+module.exports.getProducerWebsocketPulsarUrlWithTopic = function (topic) {
+    return getWebsocketPulsarUrl('producer') + `/${topic}`;
+}
+
+module.exports.getConsumerWebsocketPulsarUrlWithTopic = function (topic, subscription = '') {
```

Comment:
`susbscription` can be empty ?

Reply:
hum... no I don't think so
https://pulsar.apache.org/docs/en/client-libraries-websocket/#consumer-endpoint